lib/cocoapods/validator.rb

```diff
@@ -395,19 +395,37 @@ def validate_documentation_url(spec)
 
     # Performs validation for which version of Swift is used during validation.
     #
+    # An error will be displayed if the user has provided a `swift_version` attribute within the podspec but is also
+    # using either  `--swift-version` parameter or a `.swift-version with a different Swift version.
+    #
     # The user will be warned that the default version of Swift was used if the following things are true:
     #   - The project uses Swift at all
     #   - The user did not supply a Swift version via a parameter
+    #   - There is no `swift_version` attribute set within the specification
     #   - There is no `.swift-version` file present either.
     #
     def validate_swift_version
-      if uses_swift? && @swift_version.nil? && dot_swift_version.nil?
-        warning(:swift_version,
+      return unless uses_swift?
+      spec_swift_version = spec.swift_version
+      unless spec_swift_version.nil?
+        message = nil
+        if !dot_swift_version.nil? && dot_swift_version != spec_swift_version.to_s
+          message = "Specification `#{spec.name}` specifies an inconsistent `swift_version` (`#{spec_swift_version}`) compared to the one present in your `.swift-version` file (`#{dot_swift_version}`). " \
+                    'Please remove the `.swift-version` file which is now deprecated and only use the `swift_version` attribute within your podspec.'
+        elsif !@swift_version.nil? && @swift_version != spec_swift_version.to_s
+          message = "Specification `#{spec.name}` specifies an inconsistent `swift_version` (`#{spec_swift_version}`) compared to the one passed during lint (`#{@swift_version}`)."
+        end
+        unless message.nil?
+          error('swift', message)
```

Comment:
NIT: I think it would be cleaner to just inline this personally, but don't feel strongly about that.